packages/adaptive_scaffold/lib/src/breakpoints.dart

```diff
@@ -0,0 +1,156 @@
+  ///
+  /// It is active from a width of 840 dp to infinity.
+  static const Breakpoint large = WidthPlatformBreakpoint(begin: 840);
+
+  /// This is the defined large desktop breakpoint.
+  ///
+  /// It is active from a width of 840 dp to infinity and only on desktop
+  /// devices.
+  static const Breakpoint largeDesktop =
+      WidthPlatformBreakpoint(begin: 840, platform: _desktop);
+
+  /// This is the defined large mobile breakpoint.
+  ///
+  /// It is active from a width of 840 dp to infinity and only on mobile
+  /// devices.
+  static const Breakpoint largeMobile =
+      WidthPlatformBreakpoint(begin: 840, platform: _mobile);
+}
+
+/// A class that can be used to generate [Breakpoint]s that depend on the screen
+/// width and the platform quickly.
+class WidthPlatformBreakpoint extends Breakpoint {
+  /// Returns a [Breakpoint] with the given constraints.
+  const WidthPlatformBreakpoint({this.begin, this.end, this.platform});
+
+  /// The beginning width dp value. If left null then the [Breakpoint] will have
+  /// no lower bound.
+  final double? begin;
+
+  /// The end width dp value. If left null then the [Breakpoint] will have
+  /// no upper bound.
+  final double? end;
+
+  /// A list of [TargetPlatform]s that the [Breakpoint] will be active on. If
+  /// left null then it will be active on all platforms.
+  final List<TargetPlatform>? platform;
+
+  @override
+  bool isActive(BuildContext context) {
+    bool size = false;
+    final bool isRightPlatform =
+        platform?.contains(Theme.of(context).platform) ?? true;
+    if (begin != null && end != null) {
+      size = MediaQuery.of(context).size.width >= begin! &&
+          MediaQuery.of(context).size.width < end!;
+    } else if (begin != null && end == null) {
+      size = MediaQuery.of(context).size.width >= begin!;
+    } else if (begin == null && end != null) {
+      size = MediaQuery.of(context).size.width < end!;
+    }
+    return size && isRightPlatform;
+  }
+}
+
+/// An interface to define the conditions that distinguish between types of
+/// screens.
+///
+/// Adaptive apps usually display differently depending on the screen type: a
+/// compact layout for smaller screens, or a relaxed layout for larger screens.
+/// Override this class by defining `isActive` to fetch the screen property
+/// (usually `MediaQuery.of`) and return true if the condition is met.
+///
+/// Typically, it is not needed to keep breakpoints exclusive between each
+/// other, since they are tested one by one with a pre-defined priority.
+///
+/// If the condition is only based on the screen width and/or the device type,
+/// use [WidthPlatformBreakpoint] to define the [Breakpoint].
+///
+/// See also:
+///
+///  * [SlotLayout.config], which uses breakpoints.
+abstract class Breakpoint {
+  /// Returns a [Breakpoint].
+  const Breakpoint();
+
+  /// A method that returns whether the breakpoint is active under some
```

Comment:
Returns true if conditions based on the context, like `MediaQuery.of(context).size.width`, indicate that this breakpoint should defined the app's layout.